tests/php/ProtectedFileControllerTest.php

```diff
@@ -321,7 +321,7 @@ public function testFolders()
 
         // Flysystem reports root folder as not present
         $result = $this->get('assets');
-        $this->assertResponseEquals(404, null, $result);
+        $this->assertResponseEquals(403, null, $result);
```

Comment:
A 403 seems like a security downgrade from a 404, as it acknowledges to the attacker that this file exists, it's just that they cannot access it.  A 404 response does not lot the attacker know the file even exists.

Can we change this behavior so that a 404 is returned instead of a 403?

Reply:
Fix it by implementing old version of `Filesystem` `has()` method in `SilverStripe\Assets\Flysystem\Filesystem` class.

Reply:
DONE